src/user/user.controller.ts

```diff
@@ -0,0 +1,62 @@
+import { Controller, Get, Post, Body, Param, Delete, UsePipes, ValidationPipe, Put, HttpCode } from '@nestjs/common'
+import { UserService } from './user.service'
+import { CreateUserResponse, GetUserDto, UpdatePersonalInfoDto, UpdateRelationshipDto } from './dto/user.dto'
+import { CreateUserDto } from './dto/user.dto'
+import { ApiBadRequestResponse, ApiOkResponse, refs } from '@nestjs/swagger'
+
+@Controller('user')
+export class UserController {
+  constructor(private readonly userService: UserService) {}
+  @Get(':id')
+  @ApiOkResponse({ type: GetUserDto })
+  @ApiBadRequestResponse({ description: 'User not found' })
+  async getUser(@Param('id') id: number): Promise<GetUserDto> {
+    const user = await this.userService.findOne({ uid: id }, { shouldExist: true })
+    return this.userService.schemaToDto(user)
+  }
+  @Get('/relationship/caretaker/:eid')
+  @ApiOkResponse({ type: [GetUserDto] })
+  @ApiBadRequestResponse({ description: 'User not found' })
+  async getCaretakersByElderlyId(@Param('eid') eid: number) {
+    const users = await this.userService.findCaretakerByElderlyId(eid)
+    return users.map((user) => this.userService.schemaToDto(user))
```

Comment:
I think u should send only the required variable back to frontend don't send the whole info of user(I know u already delete sensitive data).

Reply:
Like what we discussed earlier, I'll modify this endpoint and change it to `getProfile` (only for profile page)